src/main/java/core/basesyntax/service/ReportService.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class ReportService {
+
+    public static void buildReport(String filePath, StorageService storage) {
+        File file = new File(filePath);
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(file, true))) {
+            for (Map.Entry<String, Integer> entry : storage.getStorageMap().entrySet()) {
+                writer.write(String.valueOf(entry).replace('=', ','));
+                writer.write("\n");
+                writer.flush();
```

Comment:
There's no point in using BufferedWriter if you flush() every time you write to the buffer, you can use FileWriter instead.
Piece from the documentation on the flush() method "write them immediately to their intended destination."